### PR TITLE
Allocate a timestamp when createIndexForSegment

### PR DESCRIPTION
**Pull request by wayblink**
related: #21979

Signed-off-by: wayblink <anyang.wang@zilliz.com>

### CODE REVIEW

**Comment by xiaofan-luan** [comment]
lgtm

good catch!

File: internal/datacoord/index_service.go [hunk 1]
```diff
@@ -335,7 +339,7 @@ func (s *Server) completeIndexInfo(indexInfo *indexpb.IndexInfo, index *model.In
 			}
 			continue
 		}
-		if segIdx.CreateTime > index.CreateTime {
+		if uint64(seg.GetID()) > index.CreateTime {
```
[the remark this answers is not in the document]
Comment (wayblink) [reply internal/datacoord/index_service.go:342]: we may create index after insert. So segmentID will be smaller than index.CreateTime. 
segmentIndex CreateTime > collection index CreateTime > segment CreateTime(ID)
Which means this expr will always be false.

**Pull request closed without merging**